File: C++/src/algorithms/bubble_sort/bubble_sort.tests.cpp
#include <gtest/gtest.h>
#include "bubble_sort.h"

namespace BubbleSortTests
{
    TEST(BubbleSortTests, BubbleSort_IsCorrect)
    {
        int arr[] = {9, 3, 799, 4, 420};
        int expected[] = {3, 4, 9, 420, 799};

        int arrSize = sizeof(arr) / sizeof(int);

        BubbleSort::bubble_sort(arr, arrSize);

        for (int i = 0; i < arrSize; i++)
            std::cout << arr[i] << " ";
        std::cout << std::endl;

        for (int i = 0; i < arrSize; i++)
            EXPECT_EQ(arr[i], expected[i]);
    }
}

File: CSharp/Algorithms/BubbleSort/BubbleSort.Tests.cs
using NUnit.Framework;

namespace CSharp.Algorithms.BubbleSort;

[TestFixture]
public class BubbleSortTests
{
    [Test]
    [TestCase((int[])[], (int[])[])]
    [TestCase((int[])[1], (int[])[1])]
    [TestCase((int[])[1, 3, 2], (int[])[1, 2, 3])]
    [TestCase((int[])[9, 4, 1, 6, 3, 2], (int[])[1, 2, 3, 4, 6, 9])]
    public void Assert_Array_Is_Sorted(int[] arr, int[] expected)
    {
        Assert.That(BubbleSort.Sort(arr), Is.EqualTo(expected));
    }
}
